### PR TITLE
added Lesk algorithm to nltk.sem

### PR DESCRIPTION
**Pull request by alvations**
Written a simple lesk algorithm for nltk.sem

### CODE REVIEW

**Comment by stevenbird** [comment]
Thanks for this contribution. Would you please make sure it follows our guidelines?
https://github.com/nltk/nltk/wiki/Developers-Guide

**Comment by alvations** [comment]
I have modified the lesk.py to suit the guidelines and also added the doctest. Hopefully i didn't leave out anything.

File: nltk/sem/lesk.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    """
+    Calculate overlaps between the context sentence and the synset_signature
+    and returns the synset with the highest overlap.
+    
+    :param context: ``context_sentence`` The context sentence where the ambiguous word occurs.
+    :param synsets_signatures: ``dictionary`` A list of words that 'signifies' the ambiguous word.
+    :param pos: ``pos`` A specified Part-of-Speech (POS).
+    :return: ``lesk_sense`` The Synset() object with the highest signature overlaps.
+    """
+    max_overlaps = 0; lesk_sense = None
```
Comment (kmike) [inline nltk/sem/lesk.py]: could you please split it into 2 lines?

File: nltk/sem/lesk.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    sense of each word. See http://goo.gl/8TB15w
+    
+    :param context_sentence: The context sentence where the ambiguous word occurs.
+    :param ambiguous: The ambiguous word that requires WSD.
+    :param pos: A specified Part-of-Speech (POS).
+    :param dictionary: A list of words that 'signifies' the ambiguous word.
+    :return: ``lesk_sense`` The Synset() object with the highest signature overlaps.
+    """
+    if not dictionary:
+        dictionary = {ss:ss.definition().split() \
```
Comment (kmike) [inline nltk/sem/lesk.py]: NLTK still supports Pytho 2.6, and dict comprehensions are not available there.

**Comment by alvations** [comment]
@kmike, split and un-dictionary comprehended

**Comment by stevenbird** [comment]
Thanks @alvations.

**Comment by alvations** [comment]
You're welcomed. Maybe the WSD will be a package on its own someday.

**Comment by stevenbird** [comment]
Note that I've put it in a new module `nltk/wsd.py` and called the function `lesk()`.

**Pull request merged**